### PR TITLE
callCabal2nix: Use stdenvNoCC

### PR DESCRIPTION
**Pull request by roberth**
###### Motivation for this change

Delays C compiler realisation to the actual build, which is more parallelizable; doesn't block the evaluator.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] *not testable on hydra* Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/make-package-set.nix [hunk 1]
```diff
@@ -124,20 +124,17 @@ let
   haskellSrc2nix = { name, src, sha256 ? null, extraCabal2nixOptions ? "" }:
     let
       sha256Arg = if sha256 == null then "--sha256=" else ''--sha256="${sha256}"'';
-    in buildPackages.stdenv.mkDerivation {
-      name = "cabal2nix-${name}";
+    in buildPackages.runCommand "cabal2nix-${name}" {
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/make-package-set.nix:127]: Wondering why `buildPackages` was used here in the first place…

Comment (roberth) [reply pkgs/development/haskell-modules/make-package-set.nix:127]: `git blame` reveals
 - https://github.com/NixOS/nixpkgs/pull/70192

**Comment by cdepillabout** [comment]
This LGTM.

I tested this with the following short nix file in the base of my nixpkgs checkout:

```nix
with import ./. {}; haskellPackages.hackage2nix "resourcet" "1.2.4.3"
```

Running from current `master` (3473f50affc):

```console
$ nix-build ./test.nix 
/nix/store/p99xbbc3jwb7v1bjmprzb66idry7m27r-cabal2nix-resourcet-1.2.4.3
```

Running from this PR (b4524d8eda7fae5fc81e83286a402a37526220eb):

```console
$ nix-build ./test.nix 
/nix/store/bwscp8sz4aqip2w60mv091bpvkbq9025-cabal2nix-resourcet-1.2.4.3
```

Comparing the output `.nix` files, you can see that they are the same:

```console
$ diff -u /nix/store/p99xbbc3jwb7v1bjmprzb66idry7m27r-cabal2nix-resourcet-1.2.4.3/default.nix /nix/store/bwscp8sz4aqip2w60mv091bpvkbq9025-cabal2nix-resourcet-1.2.4.3/default.nix 
$ echo $?
0
```

**Comment by cdepillabout** [comment]
Also, looking at these two derivations with `nix-diff`, it does appear that with this PR, `gcc-wrapper` is no longer pulled in.

**Pull request merged**